adr/0021-rules-for-entity-mapping-in-service-catalog.md

```diff
@@ -0,0 +1,175 @@
+
+`System` resource belongs to a one domain. `System` resource owns `Component` resources.
+
+If there are multiple instances of the same service system, either for different maturity level environments (staging/production) or intended to different audiences (ODH in Smaug and ODH in OS Climate) the System or Component resources need to be duplicated.
+
+### Kind: Component
+
+Individual service, a runtime, component. Usually describes an individual deployment. A component should map to
+
+Examples:
+
+* **ArgoCD Repository Server Component**
+* **Deck Component of Prow**
+
+In special cases of complicated architectures, where there are multiple entities in between a `Domain`, `System` and a deployment, the `Component` resource can describe a "System-like" resources. The most obvious example here is applications which belong to Open Data Hub:
+
+> Open Data Hub is described as a `System` resource in Data Science `Domain`. Due to lack of intermediate resource types in Backstage, all the applications within Open Data Hub, like Superset, JupyterHub, Argo Workflows, etc. are described as `Component` resources. These resources have subcomponents (specified as `Component` resources with the [`.spec.subcomponentOf` field](https://backstage.io/docs/features/software-catalog/descriptor-format#specsubcomponentof-optional) specified) which finally map to individual deployments
+
+* **JupyterHub Component**
+* **JupyterHub Controller subcomponent** `Component` resource
+* **Idle culler subcomponent** `Component` resource
+
+The `Component` resource requires a [`.spec.type` field](https://backstage.io/docs/features/software-catalog/descriptor-format#spectype-required) to be specified. While any arbitrary value is allowed here, we recognize following main types:
+
+* `service`: Any backend-like application, API provider, bot
+* `website`: Has to provide frontend web interface
+* `library`: Provides a library or any sort of code base dependency
+* `template`: Is used as a repository or service template
+
+### Kind: API
+
+If a component is a backend service, it may expose an API intended to be consumed by others. Each exposed API should provide an definition, either statically or dynamically. Backstage [requires us to specify](https://backstage.io/docs/features/software-catalog/descriptor-format#kind-api) `.spec.definition` and `.spec.type` fields. Definition should reflect selected type. Additionally `.spec.definition` can directly describe the api verbatim or can reference remote or service provided dynamic API definitions through supported [substitution modifiers](https://backstage.io/docs/features/software-catalog/descriptor-format#substitutions-in-the-descriptor-format).
+
+### Kind: Resource
+
+This resource type represents any other type of dependency of a `Component`. It can be an object bucket, cluster where the workload is hosted at, etc. Any external dependency that the service relies on.
```

Comment:
Maybe we can also add that the `.spec.type` filed is required and provide some common values like the `Kind: Component`?
```suggestion
This resource type represents any other type of dependency of a `Component`. It can be an object where the workload is hosted at. Any external dependency that the service relies on. It also requireds a [`.spec.type` field](https://backstage.io/docs/features/software-catalog/descriptor-format#spectype-required-4) to be specified. We recognize following main types:
* `database`: A database system that the component depends on
* `s3-bucket`: AWS S3 bucket storage
* `cluster`: OpenShift or Kuberntes cluster
```